### PR TITLE
manifest: openthread upmerge up to commit 1f1a826 

### PR DESCRIPTION
**Pull request by mariuszpos**
manifest: openthread upmerge up to commit 1f1a826

Added new configs:
CONFIG_OPENTHREAD_DNS_QUERY_UPSTREAM -> OT_DNS_QUERY_UPSTREAM
CONFIG_OPENTHREAD_MESH_DIAG -> OT_MESH_DIAG

### CODE REVIEW

**Comment by rlubos** [comment]
@mariuszpos You need to open a PR in https://github.com/zephyrproject-rtos/openthread first and use it as a revision in the manifest file. See https://github.com/zephyrproject-rtos/openthread/pull/78 for example on how the OT upmerge PR should look like.

File: modules/openthread/Kconfig.features [hunk 1]
```diff
@@ -235,6 +241,9 @@ config OPENTHREAD_SRP_SERVER
 	select OPENTHREAD_NETDATA_PUBLISHER
 	select OPENTHREAD_ECDSA
 
+config OPENTHREAD_TCP
```
Comment (canisLupus1313) [inline modules/openthread/Kconfig.features:244]: Please align it with `OPENTHREAD_TCP_ENABLE` and `zephyr/modules/openthread/platform/openthread-core-zephyr-config.h`

File: west.yml [hunk 2]
```diff
@@ -203,7 +203,7 @@ manifest:
       revision: aedcc262f93bbb1b0c2f58026911575729b7465c
       path: modules/lib/open-amp
     - name: openthread
-      revision: 242c7cc7c3a80c985476689b64cdb6d8fbcf5b6d
+      revision: pull/82/head
```
Comment (rlubos) [inline west.yml:206]: You can now use SHA: `1f1a826d1891fc99d14822abc0a1de257e9c6e9b`

File: modules/openthread/Kconfig.features (hunk 1 above)
Comment (canisLupus1313) [reply modules/openthread/Kconfig.features:244]:
```suggestion
config OPENTHREAD_TCP
```
please align with `OPENTHREAD_EXCLUDE_TCPLP_LIB`

**Review by canisLupus1313 — CHANGES_REQUESTED** [review]

**Review by rlubos — COMMENTED** [review]
@mariuszpos Please check the compliance results, the new option descriptions need rewording, otherwise looks good.

**Review by rlubos — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Pull request merged**